plugins/cookie_manager/lib/src/cookie_mgr.dart

```diff
@@ -29,6 +29,10 @@ class CookieManager extends Interceptor {
 
   /// Merge cookies into a Cookie string.
   static String getCookies(List<Cookie> cookies) {
+    // remove previous cookie with same name
+    cookies.removeWhere((cookie) =>
+        cookies.indexWhere((c) => c.name == cookie.name) <
+        cookies.indexOf(cookie));
```

Comment:
I haven't got this before, but as I look into the [RFC 6265](https://www.rfc-editor.org/rfc/rfc6265), we should probably sort cookies by longer paths too, as described in the [Section 5.4](https://www.rfc-editor.org/rfc/rfc6265#section-5.4).

Reply:
Agreed. 
Btw, in the same document, it also stated that we should sort by creation-times in case there is equal length path fields. I don't aware there is a creation-times in cookies. I am wondering how should we handle this?

Reply:
We don't, which means cookies first in should stay their position if the later one has the same length of the path.

Reply:
I have added the sorting logic and new unit test. Please review.